services/compute/virtualmachine/virtualmachine.go

```diff
@@ -144,14 +144,38 @@ func (c *client) getWssdVirtualMachineStorageConfigurationImageReference(s *comp
 	return *s.Name, nil
 }
 func (c *client) getWssdVirtualMachineStorageConfigurationOsDisk(s *compute.OSDisk) (*wssdcloudcompute.Disk, error) {
-	if s.Vhd == nil {
-		return nil, errors.Wrapf(errors.InvalidInput, "Vhd Configuration is missing in OSDisk")
+	if s.ManagedDisk == nil {
```

Comment:
Why are we adding the VHD check inside the managedisk?

Reply:
I added a comment, hopefully that makes it clearer.
If ManagedDisk == nil it's the normal VM creation from OsDisk and we want to error if Vhd is invalid.
Otherwise, we allow Vhd to be invalid since we may be creating VM from imagereference